lib/datasource/go.js

```diff
@@ -89,7 +90,9 @@ async function getPkgReleases(purl, config) {
  */
 
 async function getDigest(config) {
-  const githubPurl = await getSourcePurl(config.depName);
+  const purl = parse(config.purl);
+  const name = purl ? purl.fullname : config.depName;
+  const githubPurl = await getSourcePurl(name);
```

Comment:
@rarkins would this cause any side effects?

Reply:
Or is it better to add `purl` to the arguments of `getDigest`

Reply:
If this approach works for now, let's leave it and maybe refactor later